### PR TITLE
[libunwind][Haiku] Improve support

### PR DESCRIPTION
**Pull request by korli**
* Signal frame unwinding on x86_64 from X512
* Header search for commpage_defs.h on non-standard paths

Unwind supported tests pass on Haiku x86_64

### CODE REVIEW

**Comment by brad0** [comment]
cc @MaskRay

File: libunwind/src/UnwindCursor.hpp [hunk 1]
```diff
@@ -2917,7 +2980,8 @@ template <typename A, typename R> int UnwindCursor<A, R>::step(bool stage2) {
 
   // Use unwinding info to modify register set as if function returned.
   int result;
-#if defined(_LIBUNWIND_CHECK_LINUX_SIGRETURN)
+#if defined(_LIBUNWIND_CHECK_LINUX_SIGRETURN) ||                               \
+    (defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64))
```
Comment (MaskRay) [inline libunwind/src/UnwindCursor.hpp:2984]: Why is `_LIBUNWIND_TARGET_X86_64` check needed?

Comment (korli) [reply libunwind/src/UnwindCursor.hpp:2984]: I suppose because signal frame unwinding is only tested on x86_64. 
@X547

Comment (X547) [reply libunwind/src/UnwindCursor.hpp:2984]: riscv64 support can be added with a little effort.

Comment (brad0) [reply libunwind/src/UnwindCursor.hpp:2984]: I think you could remove the _LIBUNWIND_TARGET_X86_64 checks for all but around line 2758.

File: libunwind/src/UnwindCursor.hpp [hunk 2]
```diff
@@ -2749,6 +2755,63 @@ int UnwindCursor<A, R>::stepThroughSigReturn(Registers_arm64 &) {
   _isSignalFrame = true;
   return UNW_STEP_SUCCESS;
 }
+#elif defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64)
```
Comment (MaskRay) [inline libunwind/src/UnwindCursor.hpp:2758]:
```
}

#elif ...
#include ...
```

the blank line is preferred before `#elif`, not after

**Review by MaskRay — APPROVED** [review]

File: libunwind/src/UnwindCursor.hpp (hunk 2 above)
Comment (korli) [reply libunwind/src/UnwindCursor.hpp]: Done.

**Comment by brad0** [comment]
@X547 Please look at adding the RISCV64 support.

**Comment by trungnt2910** [comment]
This code has **not** been tested against `tests/signal_unwind.pass.cpp`, and this exact test has failed when tested against `libunwind` in LLVM 20 on Haiku.

See #135367 for an alternative implementation that works, and as a bonus, does not rely on private headers.

**Pull request merged**